### DISCUSSION TITLE
how to apply colorbox to a row or a column in matrix

### DISCUSSION THREAD

**Discussion by sophia-hye** [KaTeX/KaTeX#3785] [Q&A]
# question
I want to fill cell's background of my matrix in diagonal. I tried to use colorbox but the result is not pretty. (not aligned background..) Is there a way to do this? 
## expected result
![expected result image](https://user-images.githubusercontent.com/77262733/219335358-4e46da60-62e4-4a0d-afa2-ad7c1331d437.png)
※ I use this result on overleaf using latex and package. 
<details>
<summary>This is the latex code.</summary>
  \documentclass{article}
  \usepackage{graphicx} % Required for inserting images
  \usepackage[table]{xcolor}
  
  \newcommand\bgcell{\cellcolor{lightgray}}
  
  \begin{document}
  
  \begin{equation}
      \sum=\left(\begin{array}{cccc}
      \rowcolor{lightgray}
      \sigma_{x_1}^2 & \sigma_{x_1x_2} & \sigma_{x_1x_3} \\
      \sigma_{x_1x_2} & \bgcell \sigma_{x_2}^2 & \bgcell \sigma_{x_2x_3} \\
      \sigma_{x_1x_3} & \sigma_{x_2x_3} & \bgcell \sigma_{x_3}^2 \\
      \end{array}\right)
  \end{equation}
  
  \end{document}
</details>

# katex code
\begin{align}
\sum=\left(\begin{array}{}
\colorbox{gray}{$\sigma_{x_1}^2$} & \colorbox{gray}{$\sigma_{x_1x_2}$} & \sigma_{x_1x_3} \\[0.5em]
\sigma_{x_1x_2} & \sigma_{x_2}^2 & \sigma_{x_2x_3} \\[0.5em]
\sigma_{x_1x_3} & \sigma_{x_2x_3} & \sigma_{x_3}^2 \\
\end{array}\right)
\end{align}

# katex code result
$$
\begin{align}
\sum=\left(\begin{array}{}
\colorbox{lightgray}{$\sigma_{x_1}^2$} & \colorbox{lightgray}{$\sigma_{x_1x_2}$} & \sigma_{x_1x_3} \\
\sigma_{x_1x_2} & \sigma_{x_2}^2 & \sigma_{x_2x_3} \\
\sigma_{x_1x_3} & \sigma_{x_2x_3} & \sigma_{x_3}^2 \\
\end{array}\right)
\end{align}
$$

**Comment by edemaine** [comment]
Sadly I don't think this is possible in KaTeX proper, but it'd be great to support `\cellcolor`; see #2196.

Meanwhile, I imagine it might be possible with some CSS hacking.

**Comment by mbourne** [comment]
@sophia-hye Did you have any luck getting any CSS hack to work? I couldn't see how to do it since the \align environment uses columns and shading the spaces between the columns proved a challenge.

Here's a hacky solution (the purists will be horrified) which looks OK.

Put the matrix elements in a HTML table and surround it with parentheses:

```
<div class="shader">
  <div>
    $$ \sum=\Biggl( $$
  </div>
  <div>
    <table>
      <tr><td>$\sigma_{x_1}^2$</td><td>$\sigma_{x_1x_2}$</td><td>$\sigma_{x_1x_3}$</td></tr>
      <tr><td>$\sigma_{x_1x_2}$</td><td>$\sigma_{x_2}^2$</td><td>$\sigma_{x_2x_3}$</td></tr>
      <tr><td>$\sigma_{x_1x_3}$</td><td>$\sigma_{x_2x_3}$</td><td>$\sigma_{x_3}^2$</td></tr>
    </table>
  </div>
  <div>
    $$ \Biggr) $$
  </div>
</div> 
```

Style it as follows:

```
.shader table{border-collapse:collapse}
.shader td{text-align:center;padding:0 0.5em}
.shader tr:first-child td,
  .shader tr:nth-child(2) td:nth-child(n+2),
  .shader tr:last-child td:last-child{background:#ddd}
.shader{text-align:center}
.shader div{display:inline-block;vertical-align:middle}
.shader .delimsizing.size4{font-size:1.45em}
```

The result, with ordinary KaTeX rendering of the matrix for layout comparison:

![image](https://user-images.githubusercontent.com/7541940/221331018-2d3a7d36-7a0b-49c1-912f-8ef67eb117e5.png)

Hope it's of some use.

Regards
Murray
